code/modules/urist/mob/uristliving.dm

```diff
@@ -0,0 +1,64 @@
+//vg color-matrix code
+/mob/proc/get_screen_colour()
+	if(!client)
+		return 0
+	if(M_NOIR in mutations)
+		return NOIRMATRIX
+
+/mob/dead/observer/get_screen_colour()
+	return default_colour_matrix
+
+/mob/living/simple_animal/get_screen_colour()
+	. = ..()
+	if(.)
+		return .
+	else if(src.colourmatrix.len)
+		return src.colourmatrix
+
+/mob/living/carbon/human/get_screen_colour()
+	. = ..()
+	if(.)
+		return .
+	//var/datum/organ/internal/eyes/eyes = internal_organs_by_name["eyes"]
+	//if(eyes && eyes.colourmatrix.len && !(eyes.robotic))
+	//	return eyes.colourmatrix
+	//else return
+	return default_colour_matrix
+
+/mob/proc/update_colour(var/time = 50,var/forceupdate = 0)
+	if(!client || (client.updating_colour && !forceupdate))
+		return
+	var/list/colour_to_apply = get_screen_colour()
+	var/list/difference = difflist(client.color,colour_to_apply)
+	if(difference || !(client.color) || !istype(difference) || !difference.len)
+		client.updating_colour = 1
+		var/cached_ckey = client.ckey
+		if(forceupdate)
+			time = 0
+		else if(colour_to_apply == NOIRMATRIX)
+			time = 170
+		client.colour_transition(colour_to_apply,time = time)
+		spawn(time)
```

Comment:
_Callbacks?_ In _my BYOND?_ Is it fine if I define them in the same module as the Profunctor typeclass, or should I split them? 😱 

Jokes aside, I haven't even touched DM code for like two years, so if you come across any cases where some new functionality like this would apply, I appreciate the notification. Also I just might seriously wind up implementing a lazily-evaluated Function datum.